src/compiler/core.rs

```diff
@@ -85,12 +86,27 @@ pub fn concatenate<TItem>(mut array1: Vec<TItem>, mut array2: Vec<TItem>) -> Vec
     array1
 }
 
-pub fn insert_sorted<TItem>(array: &mut SortedArray<TItem>, insert: TItem) {
+fn identity_key_selector<TItem>(item: TItem, _: Option<usize>) -> TItem {
+    item
+}
+
+pub fn insert_sorted<TItem /*, TComparer: Comparer<&'array_or_item TItem>*/>(
+    array: &mut SortedArray<TItem>,
+    insert: TItem,
+    // compare: Comparer<&'array_or_item TItem>,
+    // compare: Comparer<&TItem, &TItem>,
+    compare: fn(&TItem, &TItem) -> Comparison,
```

Comment:
Per Slack, this is where I "inlined" the type alias `Comparer` since then it didn't seem to care about the lifetime of these `&TItem`'s